test/Calendar.test.js

```diff
@@ -6,34 +6,172 @@ const fixtures = require("./fixtures/index");
-            const response = fixtures.response;
-
-            function request() {
-                return Promise.resolve(response);
+            const Calendar = createCalendar(() => {
+                return Promise.resolve(fixtures.getCtagResponse);
+            });
+            const calendar = new Calendar(config);
+
+            return calendar
+            .getCtag()
+            .then((response) => {
+                expect(response).to.have.property("href", CALENDAR_PATH);
+                expect(response).to.have.property("name", "Default calendar");
+                expect(response).to.have.property("ctag", "http://sabre.io/ns/sync/3");
+            })
+            .catch((err) => {
+                throw err;
+            });
+        });
+
+    });
+
+    describe(".getEtags()", () => {
+
+        it("should return an array of object with etags of all events", () => {
+            const Calendar = createCalendar(() => {
+                return Promise.resolve(fixtures.getEtagsResponse);
+            });
+            const calendar = new Calendar(config);
+
+            return calendar
+            .getEtags()
+            .then((response) => {
+                expect(response).to.be.instanceof(Array);
+                expect(response).to.have.lengthOf(3);
+                expect(response[0]).to.have.property("ics");
+                expect(response[0]).to.have.property("etag");
+            })
+            .catch((err) => {
+                throw err;
+            });
+        });
+
+    });
+
+    describe(".getEvents()", () => {
+
+        it("should throw if param is not provided", () => {
+            const Calendar = createCalendar(() => {
+                return Promise.resolve(fixtures.getEventsResponse);
+            });
+            const calendar = new Calendar(config);
+
+            function getEvents() {
+                return calendar.getEvents();
             }
 
-            const Calendar = createCalendar(request);
+            expect(getEvents).to.throw(TypeError);
+        });
+
+        it("should throw if events param is not an array", () => {
+            const Calendar = createCalendar(() => {
+                return Promise.resolve(fixtures.getEventsResponse);
+            });
+            const calendar = new Calendar(config);
+
+            function getEvents() {
+                return calendar.getEvents({});
+            }
+
+            expect(getEvents).to.throw(TypeError);
+        });
+
+        it("should throw if events param is an empty array", () => {
+            const Calendar = createCalendar(() => {
+                return Promise.resolve(fixtures.getEventsResponse);
+            });
             const calendar = new Calendar(config);
 
+            function getEvents() {
+                return calendar.getEvents([]);
+            }
+
+            expect(getEvents).to.throw(TypeError);
+        });
+
+        it("should not throw if events param is an array of objects", () => {
```

Comment:
You don't need to test that something does not throw if it is correctly used. Results are way more interesting.
We are only interested in that kind of test if there is a code like`throw new Error()` in the logic.